### DISCUSSION TITLE
Prefetching from API?

### DISCUSSION THREAD

**Discussion by dvdbrk** [StopNGo/react-proto#2] [General]
Excellent template, thank you.

I am confused as to where would you put an async API call during the server side rendering?
For example to fetch the Pokemon and put it in the initial state alongside the counter value?

```
  const preloadState: Partial<RootState> = { counter: { value: 88 }};
```

**Comment by StopNGo** [comment]
Hello! Thanks  for the evaluation.

Here is how it should work:
https://redux-toolkit.js.org/rtk-query/usage/server-side-rendering#server-side-rendering-elsewhere

So, before preloadState initialization you can do:
`await Promise.all(dispatch(api.util.getRunningQueriesThunk()))`
and insert the result to the preloadState.

I will try to add prefetching example into the next update.

**Reply by OlgaPolunina** [reply]
Great template. Really, the only thing missing is the SSR for API (RTQ).
Thank you!

**Comment by StopNGo** [comment]
Hi! You can now check the latest release to see how to do API requests on server side and initialize the preloaded state correctly for SSR.
Bee free to ask me any other questions.
Thanks!
